[WIP] Adding user.watching_location() method

milaaraujo opened the pull request:
Fixes #4551 

>Start by developing a test fixture with a 2-coordinate subscription user tag, and then see if we can properly detect overlap using User.watchingLocation(lat, lon, lat2, lon2) which should return a collection of User records.

`user.watching_location(nwlat, selat, nwlng, selng)` returns all the users that are subscribed to an area **inside** (nwlat, selat, nwlng, selng)`.

jywarren commented:
Wow this was fast! And impressive! 

I'll look at resolving the codeclimate notes, but i do wonder if we could compress those very repetitive lines at all? But not a big deal.

Could we add a negative test? Where we confirm it doesn't return a given result?

Also, would you mind changing the coordinate values to be below 180, which is typically a max value for longitude, and also introducing one that's negative, since we can have negative lat/lon values as well? 

One last thing - let's note the order of coordinates in a comment in the test, since some folks may use the test as an example of how to use the code. Sound good?

Then this should be good to merge, very cool! Thanks!

milaaraujo commented:
> would you mind changing the coordinate values to be below 180, which is typically a max value for longitude

As you can see I'm not very good at geography! :joy: 

I think I addressed all your points now. And we actually don't need the "Must contain all four coordinates" check... Calling the method using less than 4 parameters would failed anyway (`wrong number of arguments`).

milaaraujo commented:
Hey @jywarren, can you review this?
